### ISSUE TITLE
glog package cannot be used (due to gflags)

### ISSUE THREAD

**Issue by kapilash** [microsoft/vcpkg#20131]
**Describe the bug**
Trying to use glog package fails with "cannot ALIAS target" error on linux.

**Environment**
- OS: Linux
- Compiler: GCC 7.5

**To Reproduce**
1. ./vcpkg install glog
2.  Use glog in the CMakeLists.txt as recommended by the tool
3. mkdir build
4. cd build
5.  Run cmake like so  `cmake .. -DCMAKE_TOOLCHAIN_FILE=/path/to/vcpkg/scripts/buildsystems/vc`
      
```
find_package(glog CONFIG REQUIRED)
target_link_libraries(main PRIVATE glog::glog)
```
Full CMakeLists.txt
```
cmake_minimum_required (VERSION 3.12)

project(gfltest VERSION 0.1 LANGUAGES CXX)
find_package(glog CONFIG REQUIRED)

if (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
 add_compile_options(-std=c++14 -stdlib=libc++ -O2)
elseif(CMAKE_CXX_COMPILER_ID MATCHES "GNU")
 add_compile_options(-std=c++14 -O2)
endif()


add_executable(gfltest Hello.cpp)
target_link_libraries(gfltest PRIVATE glog::glog)


```

Repro code when

Make file generation fails with the following error

```
-- Detecting CXX compile features - done
CMake Error at /x/xx/vcpkg/scripts/buildsystems/vcpkg.cmake:574 (_add_library):
  _add_library cannot create ALIAS target "gflags" because target
  "gflags::gflags" is imported but not globally visible.
Call Stack (most recent call first):
  /x/xx/vcpkg/installed/x64-linux/share/gflags/vcpkg-cmake-wrapper.cmake:7 (add_library)
  /x/xx/vcpkg/scripts/buildsystems/vcpkg.cmake:738 (include)
  /x/xx/vcpkg/downloads/tools/cmake-3.17.2-linux/cmake-3.17.2-Linux-x86_64/share/cmake-3.17/Modules/CMakeFindDependencyMacro.cmake:47 (find_package)
  /x/xx/vcpkg/installed/x64-linux/share/glog/glog-config.cmake:23 (find_dependency)
  /x/xx/vcpkg/scripts/buildsystems/vcpkg.cmake:784 (_find_package)
  CMakeLists.txt:6 (find_package)

```

**Expected behavior**
The cmake step should succeed allowing me to continue with build.

**Failure logs** 
Not applicable

**Additional context**
I can temporarily work around the problem by adding the keyword `GLOBAL` to the add_library command in `installed/x64-linux/share/gflags/gflags-*.cmake` 
For eg  `add_library (${GFLAGS_TARGET_PREFIX}gflags SHARED IMPORTED GLOBAL)`

**Comment by NancyLi1013** [comment]
Hi @kapilash 

Thanks for posting this issue.

This issue will be fixed via PR #20011.

**Comment by PhoebeHui** [comment]
@kapilash, the PR has been merged, please get latest source and try again.
